mlrun/feature_store/api.py

```diff
@@ -628,16 +628,18 @@ def ingest(
 
     _infer_from_static_df(df, featureset, options=infer_stats)
 
-    if isinstance(source, DataSource):
+    if isinstance(source, DataSource) and source.schedule:
         for target in featureset.status.targets:
-            if (
-                target.last_written == datetime.min
-                and source.schedule
-                and source.start_time
-            ):
-                # datetime.min is a special case that indicated that nothing was written in storey. we need the fix so
-                # in the next scheduled run, we will have the same start time
-                target.last_written = source.start_time
+            max_time = (
+                max(df[featureset.spec.timestamp_key])
+                if featureset.spec.timestamp_key
+                and featureset.spec.timestamp_key in df
+                and df[featureset.spec.timestamp_key].shape[0] != 0
+                else None
```

Comment:
`featureset.spec.timestamp_key` is long and repeated 4 times here... Maybe we can make this more readable with a shorter local identifier? E.g.
```python
timestamp_key = featureset.spec.timestamp_key
```